File: 1-1.lua
io.input('1.txt')
local current
local previous

local count = 0
for line in io.lines() do
    current = tonumber(line)
    if previous then
        if current > previous then
            count = count + 1
        end
    end
    previous = current
end

print(string.format('1.1: Count = %d', count))

File: 1-1.c
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>

#define LEN 6 // 4 digits + newline + space for '\0' terminator

int main() {
	FILE *file = fopen("1.txt", "r");
	char line[LEN];

	int current = 0;
	int previous = INT_MAX;
	int count = 0;

	while (fgets(line, LEN, file)) {
		current = atoi(line);

		if (current > previous)
			count++;

		previous = current;
	}

	printf("Count: %d\n", count);
	fclose(file);
}
